### ISSUE TITLE
automatically trigger macros in certain situations

### ISSUE THREAD

**Issue by Zhaltor** [goatcorp/suggestions#824]
### Description
Activate specified macro during certain in game events. 

### More info
I find the Fader plugin very useful at minimising interface bloat, however due to the limited functionality it is less than ideal. As the game already supports hiding interface elements using macros and chat commands, I was wondering if it would be possible to activate a specified macro when certain in game events happen, such as entering/leaving combat/duties. 

----

 - [x] I checked if this request wasn't already proposed
 - [x] I made sure this request follows Dalamud's [plugin guidelines](https://github.com/goatcorp/FFXIVQuickLauncher)

**Comment by Eternita-S** [comment]
unless such "macro" is restricted to execute only client-side command it'll not be allowed
that said, having such system for client-side only commands and various stuff like settings would be quite nice

**Comment by SupineSnail** [comment]
Yeah definitely falls into automation territory to some extent. If a plug-in author wanted this in the repo though, I imagine they would have to program a trigger system themselves for a preset list of commands and not use macros at all, or actively read macros before executing to only execute white-listed commands.
